crates/flux-middle/src/rty/mod.rs

```diff
@@ -112,6 +112,19 @@ impl AdtSortDef {
         (0..self.fields()).map(|i| FieldProj::Adt { def_id: self.did(), field: i as u32 })
     }
 
+    pub fn field_names(&self) -> &Vec<Symbol> {
+        &self.0.field_names
+    }
+
+    pub fn sort_by_field_name(&self, args: &[Sort]) -> FxIndexMap<Symbol, Sort> {
+        let mut map = FxIndexMap::default();
+        let folded = self.0.sorts.fold_with(&mut SortSubst::new(args));
+        std::iter::zip(folded.iter(), self.0.field_names.iter()).for_each(|(sort, field_name)| {
+            map.insert(*field_name, sort.clone());
+        });
+        map
```

Comment:
The following reads better in my opinion (or some variation of it)

```suggestion
        std::iter::zip(&self.0.field_names, &self.0.sorts.fold_with(&mut SortSubst::new(args)))
            .map(|(name, sort)| (*name, sort.clone()))
            .collect()
```